### ISSUE TITLE
Weird behavior when using Trainer.tune 

### ISSUE THREAD

**Issue by TinTos** [PyTorchLightning/pytorch-lightning#5587]
I have two cells in a notebook in colab:

Cell1:
```python
litmodel = LitModel(num_classes, 10.00017, len(dstrain), 64, model)
trainer = Trainer(gpus=1, auto_lr_find=True, max_epochs=epochs)
trainer.tune(litmodel, train_dataloader=dltrain, val_dataloaders=dlval)
trainer.fit(litmodel, dltrain, dlval)
```

Cell2:
```python
litmodel = LitModel(num_classes, 10.00017, len(dstrain), 64, model)
trainer = Trainer(gpus=1, auto_lr_find=True, max_epochs=epochs)
trainer.tune(litmodel, train_dataloader=dltrain, val_dataloaders=dlval)
trainer2 = Trainer(gpus=1, auto_lr_find=True, max_epochs=epochs)
trainer2.fit(litmodel, dltrain, dlval)
```

Cell2 produces much better results. The loss right at the beginning of the first epoch is about 8 for Cell1 and 0.8 for Cell2. 

This is the configure_optimizers function I use:

```python
    def configure_optimizers(self):
        optimizer = torch.optim.Adam(self.parameters(), lr= self.learning_rate)
        scheduler = {
                'scheduler': torch.optim.lr_scheduler.OneCycleLR(optimizer,
                     max_lr = self.learning_rate, 
                     total_steps = 2 * self.stepsize),
                'interval': 'step',
                'frequency': 1
        }
        return [optimizer], [scheduler]
```

pytorch-lightning version is 1.1.4.

This problem does not appear in pytorch-lightning version is 1.1.2.

---

Edit by @akihironitta: Let's discuss the below issue in #101.
```
Furthermore I get the following warning when running Cell1 right at the beginning of  trainer.fit():

> /usr/local/lib/python3.6/dist-packages/torch/optim/lr_scheduler.py:136: UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`. In PyTorch 1.1.0 and later, you should call them in the opposite order: `optimizer.step()` before `lr_scheduler.step()`.  Failure to do this will result in PyTorch skipping the first value of the learning rate schedule. See more details at https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate
  "https://pytorch.org/docs/stable/optim.html#how-to-adjust-learning-rate", UserWarning)
```

**Comment by 674106399** [comment]
same problem appears in .fit(), I'm sure that it's caused by 'interval': 'step'

**Comment by edenlightning** [comment]
mind trying to upgrade to the latest version?

**Comment by norabelrose** [comment]
I can confirm that this still happens on PyTorch Lightning 1.1.8. I get the "Detected call of lr_scheduler.step() before optimizer.step()" error and I get a crazy high initial loss value, much higher than when I just manually specify the learning rate that the LR finder suggests. My model also uses 'interval': 'step' in the lr dictionary returned by configure_optimizers, so I think it's something to do with that.

**Comment by ArvinZhuang** [comment]
> I can confirm that this still happens on PyTorch Lightning 1.1.8. I get the "Detected call of lr_scheduler.step() before optimizer.step()" error and I get a crazy high initial loss value, much higher than when I just manually specify the learning rate that the LR finder suggests. My model also uses 'interval': 'step' in the lr dictionary returned by configure_optimizers, so I think it's something to do with that.

Same here, I'm using torch 1.7.1, and PyTorch Lightning 1.1.8.  Also using 'interval': 'step'.

**Comment by Borda** [comment]
@SkafteNicki mind have look? :]

**Comment by akihironitta** [comment]
Let's discuss the warning ``UserWarning: Detected call of `lr_scheduler.step()` before `optimizer.step()`.`` issue in #101 since it can be addressed independently of the performance issue.

**Comment by Borda** [comment]
any update here? :]
